platform/representation/src/MCTRepresentation.js

```diff
@@ -159,10 +162,11 @@ define(
                         path = representation && getPath(representation),
                         uses = ((representation || {}).uses || []),
                         canRepresent = !!(path && domainObject),
+                        canEdit = !!(domainObject && domainObject.hasCapability('editor')),
                         id = domainObject && domainObject.getId(),
                         key = $scope.key;
 
-                    if (unchanged(canRepresent, id, key)) {
+                    if (unchanged(canRepresent, canEdit, id, key)) {
```

Comment:
I think this does need to be modified, but I'd change the `unchanged` check to be on `domainObject`, instead of `id`. If this were the case, the change in domain object instance would change as a consequence of being wrapped for Edit mode, and `mct-representation` should pick this up and refresh.

I wonder if we have other bugs in here around this (when navigating from an original to a link, for example)...